pkg/glob/glob.go

```diff
@@ -63,11 +68,14 @@ func glob(segs []Segment, dir string, cb func(string) bool) bool {
 	}
 
 	if len(segs) == 0 {
-		return cb(dir)
+		if info, err := os.Stat(dir); err == nil {
+			return cb(PathInfo{dir, info})
+		}
+		return true
 	} else if len(segs) == 1 && IsLiteral(segs[0]) {
 		path := dir + segs[0].(Literal).Data
-		if _, err := os.Stat(path); err == nil {
-			return cb(path)
+		if info, err := os.Stat(path); err == nil {
```

Comment:
This should use `Lstat` to avoid dropping broken symlinks.

This will also drop files that you can't stat. It is a sound strategy to handle the error, since that indicates that the filesystem is changing under our feet and there isn't much we can do.

However, it's worth calling that out as an conscious choice in a comment, since this may raise a red flag to a reader as not handling the error properly.

Reply:
Since the existing, and surrounding, code uses `Stat` I'd prefer to leave that unchanged with respect to implementing this feature. If this should be changed to `Lstat` there probably needs to be a careful review of the other paths that call `Stat` and appropriate comments added as to why one or the other is used.

Similarly, when `Readdir` fails you have the comment "// XXX Silently drop the error". But that same comment isn't present in the `Stat` error cases. Since adding such comments are orthogonal to this change I'd prefer they be done in a separate change.